File: StopWatch_oc/StopWatch_oc/WSButton.m
//
//  WSButton.m
//  StopWatch
//
//  Created by ynfMac on 2019/5/6.
//  Copyright © 2019年 ynfMac. All rights reserved.
//

#import "WSButton.h"

@implementation WSButton

- (BOOL)pointInside:(CGPoint)point withEvent:(UIEvent *)event{
    if ([super pointInside:point withEvent:event]) {
        UIBezierPath *path = [UIBezierPath bezierPathWithOvalInRect:self.bounds];
        return [path containsPoint:point];
    }
    return NO;
}

- (void)setHighlighted:(BOOL)highlighted{
    
}

@end

File: stopWatch_swift/stopWatch_swift/WSButton.swift
//
//  WSButton.swift
//  stopWatch_swift
//
//  Created by ynfMac on 2019/7/9.
//  Copyright © 2019 ynfMac. All rights reserved.
//

import UIKit

class WSButton: UIButton {
    
    override func point(inside point: CGPoint, with event: UIEvent?) -> Bool {
        
        if super.point(inside: point, with: event) {
            let path = UIBezierPath.init(ovalIn: self.bounds)
            return path.contains(point)
        }
        
        return false
    }
    
    override var isHighlighted: Bool {
        
        set {
            
        }
        get {
            return false
        }
    }
    

}
